data/event/src/commonMain/kotlin/com/adammcneilly/pocketleague/data/event/CombinedStartOctaneEventService.kt

```diff
@@ -0,0 +1,98 @@
+package com.adammcneilly.pocketleague.data.event
+
+import com.adammcneilly.pocketleague.core.models.Event
+import com.adammcneilly.pocketleague.core.models.Team
+
+/**
+ * An implementation of [RemoteEventService] that combines data from an
+ * [octaneGGEventService] and a [startGGEventService].
+ *
+ * @see [getEventById] for details on why we combined these data sources.
+ */
+class CombinedStartOctaneEventService(
+    private val octaneGGEventService: RemoteEventService,
+    private val startGGEventService: RemoteEventService,
+) : RemoteEventService {
+
+    override suspend fun getUpcomingEvents(): Result<List<Event>> {
+        return startGGEventService.getUpcomingEvents()
+    }
+
+    /**
+     * Requests an event by [eventId] from the [startGGEventService], and then looks up that same
+     * event on [octaneGGEventService], so that we can fetch all possible information about an event
+     * that we may want to display.
+     */
+    override suspend fun getEventById(eventId: String): Result<Event> {
+        val startEventResult = startGGEventService.getEventById(eventId)
+
+        return startEventResult.map { startEvent ->
+            // Look up this event on octane
+            // and map in fields that were not available from Start.GG.
+            val octaneName = startEvent.parseEventNameForOctaneGG()
+
+            val octaneEvent = octaneGGEventService.getEventByName(octaneName).getOrThrow()
```

Comment:
Upon success here, we should add the octane GG event ID to our model, so that ID can be stored in the database as well. This means that in the future, we won't need to duplicate this call to octane. 

*Unless we want to consider that octane may update their information, but for these four fields that may be unnecessary.